Test/vstte2012/BreadthFirstSearch.dfy

```diff
@@ -111,10 +111,7 @@ class BreadthFirstSearch<Vertex(==)>
         C, N, d := N, {}, d+1;
       }
 
-      assert dest in R(source, d, AllVertices) ==> dest in C by { reveal R(); }
-      assert d != 0 ==> dest !in R(source, d-1, AllVertices) by { reveal R(); }
```

Comment:
Why is it okay to remove these?

Reply:
They were intermediate assertions introduced to make it easier to prove the loop invariant. But it turned out that only one of them was necessary. The rest just made Z3 work harder.

Reply:
Thanks, I suspected as much. It's hard to tell on a quick glance which is the "important" assertion. Perhaps specification coverage mechanisms would help here. :)